habitat-baselines/habitat_baselines/config/default.py

```diff
@@ -26,30 +21,12 @@
 CONFIG_FILE_SEPARATOR = ","
 
 
-lock = threading.Lock()
-
-
 def get_config(
     config_paths: str,
     overrides: Optional[list] = None,
+    configs_dir: str = _BASELINES_CFG_DIR,
 ) -> DictConfig:
-    register_hydra_plugin(HabitatConfigPlugin)
     register_hydra_plugin(HabitatBaselinesConfigPlugin)
-
-    config_path = get_full_config_path(
-        config_paths, default_configs_dir=_BASELINES_CFG_DIR
-    )
-    # If get_config is called from different threads, Hydra might
-    # get initialized twice leading to issues. This lock fixes it.
-    with lock, initialize_config_dir(
-        version_base=None,
-        config_dir=osp.dirname(config_path),
-    ):
-        cfg = compose(
-            config_name=osp.basename(config_path),
-            overrides=overrides if overrides is not None else [],
-        )
-
-    OmegaConf.set_readonly(cfg, True)
+    cfg = _habitat_get_config(config_paths, overrides, configs_dir)
```

Comment:
Habitat-Lab's configs are included in the Baselines config by specifying corresponding Habitat-Lab input config in the defaults list. In other words the Baselines config in an extension of the Habitat-Lab config. Therefore, Lab and Baselines configs have to be read/created by the same function. To make sure that all the possible custom validations/configs updates via code in Lab's config are also applied in the Baselines' `get_config`.